client/components/types.ts

```diff
@@ -0,0 +1,82 @@
+export interface Track {
+    name: string
+    link: string
+    uri: string
+    artists: [
+        {
+            name: string
+            link: string
+        },
+    ]
+    album: {
+        name: string
+        link: string
+        uri: string
+        artists: [
+            {
+                name: string
+                link: string
+            },
+        ]
+        year: number
+        icon_link: string
+    }
+    duration_ms: number
+}
+
+export interface Playlist {
+    name: string
+    link: string
+    uri: string
+    icon_link: string
+}
+
+export interface Artist {
+    name: string
+    link: string
+    uri: string
+    icon_link: string
+}
+
+export interface Album {
+    name: string
+    link: string
+    uri: string
+    artists: [
+        {
+            name: string
+            link: string
+        },
+    ]
+    year: number
+    icon_link: string
+}
```

Comment:
Would be good to modify the types as follows:

Add a new type "SimpleArtistData". Use that type to replace track and album artist list contents.
Make track album use the album type.

These types are constructed that way in the backend so they shouldn't change independently.

Reply:
Done